stix_shifter/stix_translation/src/modules/car/car_data_mapping.py

```diff
@@ -1,110 +1,31 @@
-                "hashes.SHA-256": "sha256_hash",
-                "name": "file_name",
-                "created": "creation_time",
-                "parent_directory_ref.path": "file_path"
-            }
-        },
-        "ipv4-addr": {
-            "car_type": "flow",
-            "fields": {
-                "value": "dest_ip"
-            }
-        },
-        "ipv6-addr": None,
-        "mac-addr": {
-            "car_type": "flow",
-            "fields": {
-                "value": "mac"
-            }
-        },
-        "mutex": None,
-        "network-traffic": {
-            "car_type": "flow",
-            "fields": {
-                "start": "start_time",
-                "end": "end_time",
-                "src_ref.value": "src_ip",  # Impossible to say whether this is IP, domain, or MAC
-                "dst_ref.value": "dest_ip",
-                "src_port": "src_port",
-                "dst_port": "dest_port",
-                "protocols[*]": "protocol",
-                "src_payload_ref.payload_bin": "content"
-            }
-        },
-        "process": {
-            "car_type": "process",
-            "fields": {
-                "pid": "pid",
-                "name": "exe",
-                "cwd": "current_directory",
-                "command_line": "command_line",
-                "creator_user_ref.account_login": "user",
-                "binary_ref.name": "exe",
-                "binary_ref.parent_directory_ref.path": "image_path",
-                "binary_ref.hashes.MD5": "md5_hash",
-                "binary_ref.hashes.SHA1": "sha1_hash",
-                "binary_ref.hashes.SHA256": "sha256_hash",
-                "parent_ref.name": "parent_exe",
-                "parent_ref.pid": "ppid",
-                "parent_ref.binary_ref.file_name": "parent_exe",
-                "parent_ref.binary_ref.parent_directory_ref.path": "parent_image_path",
-                "extensions.windows-process-ext.owner_sid": "sid"
-
-            }
-        },
-        "software": None,
-        "url": None,
-        "user-account": {
-            "car_type": "user_session",
-            "fields": {
-                "user_id": "logon_id",
-                "account_login": "user",
-                "account_type": "logon_type"
-            }
-        },
-        "windows-registry-key": {
-            "car_type": "registry",
-            "fields": {
-                "key": "key",
-                "values[*]": "value",
-                "creator_user_ref.account_login": "user",
-            }
-        },
-        "x509-certificate": None
-    }
+class CarDataMapper(BaseDataMapper):
 
     def __init__(self, options={}):
-        # use user defined mapping object if provided in options
-        if 'mapping' in options:
-            self.MAPPINGS = options['mapping']
+        self.options = options
+        mapping_json = options.get('mapping')
+        self.MAPPINGS = mapping_json or self.fetch_mapping()
+
+    def fetch_mapping(self):
```

Comment:
is * fetch_mapping()* used outside of this class?
you have many copies of this code..
maybe it could go to * BaseDataMapper* as a default implementation and filename parameter?

Reply:
Yeah, I guess if any of the mappers needed a different implementation of it they would just override it.